### PR TITLE
chore(mojaloop/#3522): upgrade nodejs version

### PR DESCRIPTION
**Pull request by aaronreynoza**
chore(mojaloop/#3522): nodejs upgrade - https://github.com/mojaloop/project/issues/3522

*  Upgraded NodeJS from v16 to v18 LTS
*  Updated CI config to use NodeJS latest LTS version
*  Updated CI config to display runtime versions for each job
*  Upgraded NPM dependencies
*  Fixed CI bugs caused by the NodeJS upgrade
* Updated CI pipeline use NVM to select node version from .nvmrc file
*  Upgraded circleci orbs
*  Migrated master branch to main
*  Optimized ci workflow to use cached image for some jobs
* Upgraded Jest test which needed upgrade
* Fix Husky hook

### CODE REVIEW

**Review by kleyow — APPROVED** [review]

**Review by oderayi — CHANGES_REQUESTED** [review]
Looking good!

File: package.json [hunk 1]
```diff
@@ -98,93 +100,94 @@
...
+    "eslint-plugin-import": "2.28.1",
     "eslint-plugin-node": "11.1.0",
-    "eslint-plugin-prettier": "^4.0.0",
-    "eslint-plugin-promise": "6.0.0",
+    "eslint-plugin-prettier": "^5.0.0",
+    "eslint-plugin-promise": "6.1.1",
     "eslint-plugin-standard": "4.1.0",
-    "husky": "^5.2.0",
-    "jest": "^26.6.3",
+    "husky": "^8.0.3",
```
Comment (oderayi) [inline package.json:172]: Husky is updated but the git hooks are no longer being executed

Comment (aaronreynoza) [reply package.json:172]: it does trigger for me. let me verify

Comment (oderayi) [reply package.json:172]: You can check for similar update here https://github.com/mojaloop/ml-testing-toolkit-shared-lib/pull/13

File: package.json [hunk 2]
```diff
@@ -98,93 +100,94 @@
...
+    "lint-staged": "^14.0.1",
     "multi-file-swagger": "^2.3.0",
-    "nodemon": "^2.0.18",
-    "npm-audit-resolver": "^3.0.0-7",
-    "npm-check-updates": "^14.1.1",
-    "openapi-response-validator": "^12.0.0",
-    "openapi-typescript": "^5.4.0",
-    "prettier": "^2.7.1",
+    "nodemon": "^3.0.1",
+    "npm-audit-resolver": "^3.0.0-RC.0",
```
Comment (oderayi) [inline package.json:180]: This should be removed since we're now using `audit-ci`

**Review by oderayi — CHANGES_REQUESTED** [review]

**Review by oderayi — APPROVED** [review]
+1

**Pull request merged**